### PR TITLE
[ROCm] Fix for a bug in ROCm batchnorm implementation.

### PR DESCRIPTION
**Pull request by deven-amd**
Fixing the bug also makes the unit-test `//tensorflow/python/keras/layers:normalization_test_gpu` pass, so removing the `no_rocm` tag from it as well.

------------------------------------

/cc @cheshire @chsigg @nvining-work

### CODE REVIEW

**Review by cheshire — APPROVED** [review]

**Comment by deven-amd** [comment]
@gbaned gentle ping...seems like this PR got stuck in the merge pipeline :)
